### ISSUE TITLE
How to create block types that don't fill the whole voxel cube?

### ISSUE THREAD

**Issue by modmatrix** [maxogden/voxel-engine#102]
I'd like to have block types whose geometry only takes up _part_ of a voxel - like the glass panes, slabs, and stairs in Minecraft.
1. Is that currently possible? Do I need a special module? If so, which one?

That's my main question, but here are some bonus questions if you're feeling generous:
1. If the answer to 1 is "yes": will my in-game avatar collide with the visible edges of the block, or with the edge of the voxel? For example, if I create a wall of glass panes, will I be able to move partway into that voxel (until I hit the actual glass), or will I hit an invisible wall (the edge of the voxel) just in _front_ of the glass?
2. If the answer to 2 is "you won't collide with the edge of the voxel": will the player movement routines handle these blocks in a sensible way? For example, if I create a ramp or stair block type, and I run straight into one, will I climb it? 

Basically, I'd like to create a game world that allows you to move from lower locations to higher ones without having to jump constantly.
